File: 백준/단계 별로 풀어보기/브루트_포스/1018_체스판_다시_칠하기.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <cmath>
using namespace std;

int main()
{
	cin.tie(NULL);
	cout.tie(NULL);
	ios_base::sync_with_stdio(false);
	
	int vertical, horizontal, result = 99999999;
	vector<string> board;
	
	cin >> vertical >> horizontal;
	for (int i = 0; i < vertical; i++){
	    string temp;
	    cin >> temp;
	    board.push_back(temp);
	}
	
	for (int start_y = 0; start_y < vertical - 7; start_y++){
	    for (int start_x = 0; start_x < horizontal - 7; start_x++){
	        int w_count = 0, b_count = 0;
	        for (int y = start_y; y < start_y + 8; y++){
	            for (int x = start_x; x < start_x + 8; x++){
	                if ((y & 1 && x & 1) || !(y & 1 || x & 1)){
	                    if (board[y][x] == 'W') b_count++;
	                    else w_count++;
	                }
	                
	                else{
	                    if (board[y][x] == 'W') w_count++;
	                    else b_count++;
	                }   
	            }
	        }
	        result = min(result, min(w_count, b_count));
	    }
	}
	cout << result;
}

File: baekjoon/단계 별로 풀어보기/브루트_포스/1018_체스판_다시_칠하기.py
import sys
input = sys.stdin.readline
IP = lambda: map(int, input().split())

def solve():
    vertical, horizontal = IP()
    chess_in = [input().rstrip() for i in range(vertical)]
    result = float("inf")
    
    for start_y in range(vertical - 7):
        for start_x in range(horizontal - 7):
            w_count = 0; b_count = 0
            for y in range(start_y, start_y + 8):
                for x in range(start_x, start_x + 8):
                    pl = (y & 1 == x & 1)
                    if chess_in[y][x] == 'W':
                        w_count += not pl
                        b_count += pl
                    else:
                        w_count += pl
                        b_count += not pl
            result = min(result, w_count, b_count)
    
    print(result)
    
if "__main__" == __name__: solve()